runtime/src/append_vec.rs

```diff
@@ -207,23 +203,40 @@ impl AppendVec {
         accounts
     }
 
+    #[allow(clippy::mutex_atomic)]
+    pub fn append_accounts(&self, accounts: &[(StorageMeta, &Account)]) -> Vec<usize> {
```

Comment:
@aeyakovenko, this could use some unit-tests.  At a glance, looks like it can be two functions, where the outer one is a TakeWhile, and can probably be faster by retuning a `impl Iterator` instead of `Vec`.